content/advent-2015/test2doc-generate-your-api-docs.md

```diff
@@ -0,0 +1,294 @@
++++
+author = ["Sarah Adams"]
+date = "2015-12-14T22:04:31-08:00"
+series = ["Advent 2015"]
+title = "test2doc: Generate Your API Docs"
+
++++
+
+API docs are important. We all know that.
+They are also painful and tedious to maintain.
+When your docs aren't accurate, you get more questions. And everyone 
+loses time.
+
+I've always thought of unit tests as a great source of documentation. 
+But the non-Go engineers consuming my API don't often agree.
+
+Yet most of the same information is there:
+
+- request format:
+    - HTTP method
+    - URI
+    - query params
+    - request headers
+    - request body
+- response format:
+    - status code
+    - response headers
+    - response body
+
+<br>
```

Comment:
I put them in for formatting/readability. There isn't enough spacing between sections by default IMO.
I can take them out if you prefer, just let me know.